draft-ietf-ppm-dap.md

```diff
@@ -1099,84 +1099,170 @@ refinement process. We instead think of these as properties of the output
-           requires at least two more rounds.
-       I - Produce first-round prep-share.
-       P - Combine prep shares into prep message.
-       N - Produce next prep share.
-       F - Produce output share.
-~~~
-{: #agg-flow title="Mapping of VDAF preparation to the DAP aggregation sub-protocol."}
-
-The exact protocol structure depends on the number of rounds of preparation
-that are required by the VDAF. (The number of rounds is a constant parameter
-specified by the VDAF itself: Prio3 has just one round, but Poplar1 requires
-two.) The high-level idea is that the Aggregators take turns running the
-computation locally until input from their peer is required:
+the Helper's report share for each report in the job, and for each report the
+initialization step for preparation. The Helper's response, along with each
+subsequent request and response, carries the remaining messages exchanged
+during preparation.
+
+~~~ ladder
+  report, agg_param
+   |
+   v
++--------+                                         +--------+
+| Leader |                                         | Helper |
++--------+                                         +--------+
+   | AggregationJobInitReq                               |
+   | agg_param, report_share, prep_step(initialize)      |
+   |---------------------------------------------------->|
+   |                                  AggregationJobResp |
+   |                          prep_step(continue|finish) |
+   |<----------------------------------------------------|
+   | AggregationJobContinueReq                           |
+   | prep_step(continue|finish)                          |
+   |---------------------------------------------------->|
+   |                                  AggregationJobResp |
+   |                          prep_step(continue|finish) |
+   |<----------------------------------------------------|
+   |                                                     |
+  ...                                                   ...
+   |                                                     |
+   | AggregationJobContinueReq                           |
+   | prep_step(continue|finish)                          |
+   |---------------------------------------------------->|
+   |                                  AggregationJobResp |
+   |                          prep_step(finish|finished) |
+   |<----------------------------------------------------|
+   |                                                     |
+   v                                                     v
+  leader_out_share                         helper_out_share
+~~~
+{: #agg-flow title="Overview of the DAP aggregation sub-protocol."}
+
+Each of these "preparation steps", or "prep steps" for short, has an associated
+type: one of "initialize", "continue", "finish", or "finished". The type of
+each step, and how many steps there are, depends on the VDAF. The message
+structure and processing rules are specified in {{preparation-messages}}.
```

Comment:
I don't understand what this is trying to say. These are arguments to some `prep_step()` function? How does the caller know what argument to provide? Why can't the encapsulated VDAF just do the right thing?

I'm deliberately not referring to the VDAF spec here, because the idea here is that you should be able to read this bit and not know that.

Reply:
You call `DAPPrepInit()` and `DAPPrepNext()` to do the right thing, which are defined in this document. These are actually wrapper functions around the VDAF API (i.e., `VDAF.prep_init()`, `VDAF.prep_next()`, and `VDAF.prep_shares_to_prep()`) that are specific to the 2-Aggregator, Leader-Helper structure of DAP. (VDAFs still support >2 Aggregators.)

I think we have to specify these wrapper methods somewhere. I'd be fine moving them to VDAF if you think that would be better.